### PR TITLE
[GCP] Support machine image

### LINKED ISSUE #2275: Support Machine Image on GCP
[Machine Image](https://cloud.google.com/compute/docs/machine-images) on GCP is an easy-to-use functionality that lets a user create an image from a VM. I believe this is one of the few functions that GCP does better compared to AWS. Supporting this should benefit many GCP users!

### PR DESCRIPTION
**Pull request by Michaelvll**
Fixes #2275 

```
resources:
  cloud: gcp
  image_id: projects/my-project/global/machineImages/test-machine-image
```
or
```
sky launch -c test-machine-image-gcp --cloud gcp --image-id projects/my-project/global/machineImages/test-machine-image
```

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch -c test-machine-image-gcp --cloud gcp --image-id projects/my-project/global/machineImages/test-machine-image --cpus 2+` where the machine image is in us; check the files on the VM
  - [x] `sky launch -c test-machine-image-gcp --cloud gcp --image-id projects/my-project/global/machineImages/test-machine-image --cpus 2+ --region europe-central2` where the machine image is in us; check the files on the VM
  - [x] `sky launch -c test-normal-image-gcp-2 --cloud gcp --image-id projects/deeplearning-platform-release/global/images/common-gpu-v20230615-debian-11-py310 --cpus 2+`
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Comment by zhuohan123** [comment]
Thank you @Michaelvll for the speedy fix!

File: sky/clouds/gcp.py [hunk 1]
```diff
@@ -287,9 +294,18 @@ def get_image_size(self, image_id: str, region: Optional[str]) -> float:
                 raise ValueError(f'Image {image_id!r} not found in GCP.')
             project = image_attrs[1]
             image_name = image_attrs[-1]
-            image_infos = compute.images().get(project=project,
-                                               image=image_name).execute()
-            return float(image_infos['diskSizeGb'])
+            if self._is_machine_image(image_id):
+                image_infos = compute.machineImages().get(
```
Comment (concretevitamin) [inline sky/clouds/gcp.py:297]: Probably should add an inline comment here about: We support both GCP's Machine Images and Custom Images, both of which are specified with the `image_id` field.

Should we also add to https://skypilot.readthedocs.io/en/latest/reference/yaml-spec.html ?

File: sky/clouds/gcp.py [hunk 2]
```diff
@@ -393,6 +408,10 @@ def make_deploy_resources_variables(
         assert image_id is not None, (image_id, r)
         resources_vars['image_id'] = image_id
 
+        if self._is_machine_image(image_id):
+            resources_vars['machine_image'] = image_id
```
Comment (concretevitamin) [inline sky/clouds/gcp.py:412]: Should we add `'machine_image': None` to L347?

**Review by concretevitamin — APPROVED** [review]
LGTM, thanks @Michaelvll!

File: sky/clouds/gcp.py (hunk 2 above)
Comment (Michaelvll) [reply sky/clouds/gcp.py:413]: Yes! Just discovered concurrently lol. Thansk!

File: sky/clouds/gcp.py (hunk 1 above)
Comment (Michaelvll) [reply sky/clouds/gcp.py:301]: Good point! Added. Thanks!

**Review by cblmemo — APPROVED** [review]
Just went through the code. LGTM 🙂

**Pull request merged**